### PR TITLE
Tweaked descriptions – Part 8

### PR DESCRIPTION
**Pull request by 4Luke4**
This will mostly deal with the `PRO` file format and some `2DA` files.

### CODE REVIEW

**Comment by 4Luke4** [comment]
@lynxlynxlynx 

Also, I think that <code> tags are still not sufficiently marked as such.

As you can see, I set `background-color: white` and `color: fuchsia` in `_general.scss`: what do you think?

File: _opcodes/op012-bgee.html [hunk 1]
```diff
@@ -83,6 +83,17 @@
...
+<p>
+  <code>Damage Amount</code> can be positive or negative (though the damage sound will play in either case).
+  <br/>If <code>Damage Amount</code> and the Dice values are both non-zero, the total damage inflicted is: <code>'Damage Amount' + Dice value</code>.
+</p>
+
+{% capture note %}
+<code>STUNNING</code> damage does not stun the target. <code>STUNNING</code> damage cannot kill a creature, creatures are reduced to 1 HP and are knocked unconscious.
+{% endcapture %}
+
+{% include note.html %}
```
Comment (lynxlynxlynx) [inline _opcodes/op012-bgee.html:95]: This is true for all version of this opcode.

File: _opcodes/op192-bgee.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+param2: "Unused"
+bg1: 0
+bg2: 0
+bgee: 1
+iwd1: 0
+iwd2: 0
+pst: 0
+---
+Applies <a href="#op195">opcode #195</a>, <a href="#op196">opcode #196</a>, and <a href="#op268">opcode #268</a> to the summoned familiar.<br/>
+Familiar to be determined by <a href="../files/2da/2da_bgee/familiar.htm">FAMILIAR.2DA</a> and caster's alignment.
```
Comment (lynxlynxlynx) [inline _opcodes/op192-bgee.html:14]: This is the same in bg2, so just rather merge and add a note that in EEs, the mapping has been externalized to a table.

File: _opcodes/op195-bgee.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	<li>3 &longrightarrow; Player4</li>
+	<li>4 &longrightarrow; Player5</li>
+	<li>5 &longrightarrow; Player6</li>
+</ul>
+
+{% capture note %}
+<code>Parameter2</code> is possibly intended to be <code>Damage Type</code>. It uses the same format as <a href="#op12">opcode #12</a> (<a href="#op192">opcode #192</a> sets it to <code>0x400000 – MAGIC</code>), but changing it has no effect.
+{% endcapture %}
+
+{% include note.html %}
```
Comment (lynxlynxlynx) [inline _opcodes/op195-bgee.html:37]: This can replace the main 195; I don't see anything new.

**Comment by lynxlynxlynx** [comment]
Can you post a screenshot? Sounds a bit extreme.

Things like this are silly:
```
<div class="idsHeader"><a name="BIT0">1 IDENTIFIED</a></div>
<br />
```
We can just add a margin to idsHeader or create a new class for it. Then the extra br tag will not be needed any more.

File: _opcodes/op195-bgee.html (hunk 3 above)
Comment (4Luke4) [reply _opcodes/op195-bgee.html]: Done.

File: _opcodes/op192-bgee.html (hunk 2 above)
Comment (4Luke4) [reply _opcodes/op192-bgee.html:14]: Done.

**Comment by 4Luke4** [comment]
> Can you post a screenshot? Sounds a bit extreme.

A screenshot of what...?

> Things like this are silly:
>
> ```
> <div class="idsHeader"><a name="BIT0">1 IDENTIFIED</a></div>
> <br />
> ```

Are you talking about `invitem.htm`? Well, I did that because of op236/243 (they contain references to this file...)

File: _opcodes/op192-bgee.html (hunk 2 above)
Comment (lynxlynxlynx) [reply _opcodes/op192-bgee.html:14]: You still left the file in.

File: _opcodes/op192.html [hunk 4]
```diff
@@ -10,15 +10,38 @@
...
-<br/>Lawful Neutral  Ferret (FAMFER)
-<br/>Lawful Evil     Imp (FAMIMP)
-<br/>Neutral Good    Pseudo Dragon (FAMPSD)
-<br/>True Neutral    Rabbit (FAMRAB)
-<br/>Neutral Evil    Dust Mephit (FAMDUST)
-<br/>Chaotic Good    Fairy Dragon (FAMFAIR)
-<br/>Chaotic Neutral Cat (FAMCAT)
-<br/>Chaotic Evil    Quasit (FAMQUAS)
+Applies <a href="#op195">opcode #195</a>, <a href="#op196">opcode #196</a>, and <a href="#op268">opcode #268</a> to the summoned familiar.<br/>
+On non-EE games, the Familiar summoned is based on the alignment of the Protagonist, and values stored in the <a href="../file_formats/ie_formats/gam_v2.0.htm">GAM</a> file.
```
Comment (lynxlynxlynx) [inline _opcodes/op192.html:14]:
```suggestion
The Familiar summoned is based on the alignment of the Protagonist and values stored in the <a href="../file_formats/ie_formats/gam_v2.0.htm">GAM</a> file (non-EE games) or <a href="../files/2da/2da_bgee/familiar.htm">FAMILIAR.2DA</a>.
```

File: _opcodes/op192.html [hunk 5]
```diff
@@ -10,15 +10,38 @@
...
+	<li>Neutral Evil &longrightarrow; Dust Mephit (<code>FAMDUST</code>)</li>
+	<li>Chaotic Good &longrightarrow; Fairy Dragon (<code>FAMFAIR</code>)</li>
+	<li>Chaotic Neutral &longrightarrow; Cat (<code>FAMCAT</code>)</li>
+	<li>Chaotic Evil &longrightarrow; Quasit (<code>FAMQUAS</code>)</li>
+</ul>
+
+On EE games, the Familiar summoned is determined by <a href="../files/2da/2da_bgee/familiar.htm">FAMILIAR.2DA</a> and caster's alignment.
+<ul>
+	<li>The familiars (<a href="../file_formats/ie_formats/cre_v1.htm">CRE</a>) available are locked in once a new game begins (any changes to <a href="../files/2da/2da_bgee/familiar.htm">FAMILIAR.2DA</a> must occur before starting a new game).</li>
+</ul>
```
Comment (lynxlynxlynx) [inline _opcodes/op192.html:35]:
```suggestion
{% capture note %}
The familiars (<a href="../file_formats/ie_formats/cre_v1.htm">CRE</a>) available are locked in once a new game begins (any changes to <a href="../files/2da/2da_bgee/familiar.htm">FAMILIAR.2DA</a> must occur before starting a new game).
{% endcapture %}

{% include info.html %}
```

**Comment by lynxlynxlynx** [comment]
A screenshot of your pink code tags.

Part 2: I'm just talking about the needless <br />.

**Comment by 4Luke4** [comment]
> A screenshot of your pink code tags.

Mmm, I can't see any difference (i.e., `code` is not written in fuchsia on a white background 😕 ...)
Do you have any idea?

Also, I noticed there's another gem to install in order to get the URL, i.e.: `webrick`
I'll add it to the README...

**Comment by lynxlynxlynx** [comment]
Your scss modification changed the code class, not the code tag, that's why you don't see changes when looking at the latter. Remove the starting dot and it should work. That class is rarely used:
```
file_formats/ie_formats/mus.htm:23:        <div class="code">
file_formats/ie_formats/mus.htm:70:      <div class="code">
files/ids/bg1/specific.htm:8:The file is extensible, and new entries can be added for example, adding <span class="code">55 MYSPECIFIC</span> will allow MYSPECIFIC to be used in scripts.<br />
files/ids/bg2/specific.htm:8:The file is extensible, and new entries can be added for example, adding <span class="code">55 MYSPECIFIC</span> will allow MYSPECIFIC to be used in scripts.<br />
files/ids/bg2/time.htm:11:<span class="code">4 MYTIME</span> will allow MYTIME to be used in scripts,
files/ids/bgee/time.htm:19:<span class="code">4 MYTIME</span> will allow MYTIME to be used in scripts,
files/ids/iwd/specific.htm:8:The file is extensible, and new entries can be added for example, adding <span class="code">55 MYSPECIFIC</span> will allow MYSPECIFIC to be used in scripts.<br />
files/ids/iwd2/specific.htm:8:The file is extensible, and new entries can be added for example, adding <span class="code">55 MYSPECIFIC</span> will allow MYSPECIFIC to be used in scripts.<br />
```

What do you mean with URL & webrick?

**Comment by 4Luke4** [comment]
> What do you mean with URL & webrick?

After executing `jekyll serve --incremental`, I got the following error:
`<internal:/usr/local/Cellar/ruby/3.0.0_1/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in require': cannot load such file -- webrick (LoadError)`

After installing that gem (`gem install --user-install webrick`), I successfully got the URL:
`Server address: http://127.0.0.1:4000/iesdp/`

**Comment by 4Luke4** [comment]
@lynxlynxlynx 

Some screenshots:
fuchsia

fuchsia 2

Maybe `gray` is better...?
gray

**Comment by lynxlynxlynx** [comment]
Yeah, gray is way less eyestabbing.

Ok, so webrick is perhaps needed with ruby 3 (just released). We don't support it yet. Are you sure you're not mixing ruby versions?

**Comment by 4Luke4** [comment]
> Ok, so webrick is perhaps needed with ruby 3 (just released). We don't support it yet. Are you sure you're not mixing ruby versions?

I'm using the latest version of ruby. If I execute `ruby -v`, I got:
`ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-darwin19]`

**Comment by lynxlynxlynx** [comment]
Then it sounds like a (packaging) bug of ruby on macos, since it's the ruby core that wants this plugin. We can add a note to the README, but we definitely don't need to require the install.

**Comment by 4Luke4** [comment]
> Then it sounds like a (packaging) bug of ruby on macos, since it's the ruby core that wants this plugin. We can add a note to the README, but we definitely don't need to require the install.

See the updated `README.md` and tell me if it's OK for you...

**Comment by lynxlynxlynx** [comment]
Not sure how the hr will be rendered, but the text is fine — thanks.

**Comment by 4Luke4** [comment]
> Yeah, gray is way less eyestabbing.

I also decided to remove background color, it's hopefully even better now...

Anyway, do you know if the `--incremental` flag is somewhat buggy?
Sometimes when I run `jekyll serve --incremental` and click on the generated link, it says _"Last update: 31.12.2020"_ instead of current date... Moreover, some files do not get updated upon refreshing the site...

**Comment by 4Luke4** [comment]
@burner1024 

Could you tell me why my notes about `RES` actions (see for instance `31-2.yml`) do not get displayed?

**Comment by lynxlynxlynx** [comment]
Nothing comes to mind no. If you mounted your filesystem with some of the timing flags disabled, it probably wouldn't work at all. I guess you'll have to keep restarting the server. I suggest you check and open an issue on the jekyll repo.

RES: it's an alias, so if you want notes, add it to the main action.

**Comment by 4Luke4** [comment]
As far as 2DA files are concerned, I'm using standard links for now (i.e., `<a name>`).

Do you know how to properly format (align) the page? It's currently looking bad (see below...)

<details>
Align text
</details>

**Comment by lynxlynxlynx** [comment]
I guess this was using pre before and just normal spaces to align. Tables are the next old school option and probably the simplest (you can disable the border and hardly anyone will know) ... however it's not really mobile friendly. The modern way would be to use divs for lines and a flex layout with justify-content set to space-between (div per cell). So quite verbose, let's just use tables like elsewhere.

**Comment by 4Luke4** [comment]
> I guess this was using pre before and just normal spaces to align.

Just to clarify: source code is properly aligned
```
2DA V1.0
0
                                                                                                                               StrRef
<b><a name="STRREF_ATTENTION_DIALOG">STRREF_ATTENTION_DIALOG                                                                   10467</a></b>
<b><a name="STRREF_AUTOPAUSE_ATTACKED">STRREF_AUTOPAUSE_ATTACKED                                                               17114</a></b>
<b><a name="STRREF_AUTOPAUSE_BADLYWOUNDED">STRREF_AUTOPAUSE_BADLYWOUNDED                                                       17116</a></b>
```
just like the original file, it's the rendered file that loses proper formatting and I can't understand why...

**Comment by lynxlynxlynx** [comment]
I'm not sure the html code tag preserves spaces. For non-inline use the docs specifically mention you need pre as well:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/code

Now, looking at another user of that triplet, github did generate also pre:
https://github.com/gemrb/gemrb/tree/master/demo/data#shortcut-for-running-from-the-build-dir

Are you sure you didn't accidentally use tabs?

**Comment by 4Luke4** [comment]
> I'm not sure the html code tag preserves spaces.

Actually, I'm using the `<pre>` tag, i.e.:
```
<pre>
2DA V1.0
0
                                                                                                                               StrRef
<b><a name="STRREF_ATTENTION_DIALOG">STRREF_ATTENTION_DIALOG                                                                   10467</a></b>
<b><a name="STRREF_AUTOPAUSE_ATTACKED">STRREF_AUTOPAUSE_ATTACKED                                                               17114</a></b>
</pre>
```

And as I've already told you, the final result is a mess...

> Are you sure you didn't accidentally use tabs?

I have `Select Indentation` set to `Spaces: 2`...

**Comment by lynxlynxlynx** [comment]
If you're using HTML pre and not the Markdown triplet, then you just have to fix the amount of space to fit. Have you already pushed this file?

**Comment by 4Luke4** [comment]
> If you're using HTML pre and not the Markdown triplet, then you just have to fix the amount of space to fit. Have you already pushed this file?

I pushed `STATDESC.2DA` and the problem is there too... Could you have a look at it?

File: files/2da/2da_bgee/statdesc.htm [hunk 6]
```diff
@@ -8,204 +8,218 @@
...
+<b><a name="1">1</a></b>           17392          ****
+<b><a name="2">2</a></b>           17393          ****
+<b><a name="3">3</a></b>           17394          ****
+<b><a name="4">4</a></b>           17395          ****
+<b><a name="5">5</a></b>           17396          ****
+<b><a name="6">6</a></b>           17397          ****
+<b><a name="7">7</a></b>           54337          ****
+<b><a name="8">8</a></b>           17399          ****
+<b><a name="9">9</a></b>           17400          ****
+<b><a name="10">10</a></b>           17401          ****
```
Comment (lynxlynxlynx) [inline files/2da/2da_bgee/statdesc.htm:27]: You can see here you used the same amount of space between both columns, but the double digit numbers need one less for both to align. Same with three digit numbers later.

Comment (4Luke4) [reply files/2da/2da_bgee/statdesc.htm:27]: Can that fix be automated? Or am I supposed to do that manually?

Comment (lynxlynxlynx) [reply files/2da/2da_bgee/statdesc.htm:27]: Does your editor support regular expressions? Anyway, you're on mac, so should have sed and can try this in the terminal:
`sed -ir -e 's,(>..</a></b>) ,\1,' -e 's,(>...</a></b>)  ,\1,' files/2da/2da_bgee/statdesc.htm`

Comment (4Luke4) [reply files/2da/2da_bgee/statdesc.htm:27]: > Does your editor support regular expressions? Anyway, you're on mac, so should have sed and can try this in the terminal:
> `sed -ir -e 's,(>..</a></b>) ,\1,' -e 's,(>...</a></b>) ,\1,' files/2da/2da_bgee/statdesc.htm`

Yes, my editor support regex (VSCode).
Got this error:
```
sed: 1: "s,(>..</a></b>) ,\1,
": \1 not defined in the RE
```

I'm also trying messing with `style` by setting `text-align` and/or `white-space`, but still nothing...
Something like `<pre style="text-align: justify; white-space: pre-wrap;">`...

Comment (lynxlynxlynx) [reply files/2da/2da_bgee/statdesc.htm:27]: That's suspicious, but try: `sed -i -e 's,\(>..</a></b>\) ,\1,' -e 's,\(>...</a></b>\) ,\1,' files/2da/2da_bgee/statdesc.htm`

or with the individual expressions in your editor's search&replace.

The whole point of pre is to keep things PREformatted, so styling should be limited, relayouting excluded.

Comment (4Luke4) [reply files/2da/2da_bgee/statdesc.htm:27]: > That's suspicious, but try: `sed -i -e 's,\(>..</a></b>\) ,\1,' -e 's,\(>...</a></b>\) ,\1,' files/2da/2da_bgee/statdesc.htm`
> 
> or with the individual expressions in your editor's search&replace.
> 
> The whole point of pre is to keep things PREformatted, so styling should be limited, relayouting excluded.

Now I got
`-e: No such file or directory`

Anyway, I'll go regex replacement route in VSCode...

Comment (lynxlynxlynx) [reply files/2da/2da_bgee/statdesc.htm:27]: I guess you have a bsd sed or something. The individual search expressions are between the 2nd and 3rd commas, while the replacement is just \1.

Comment (4Luke4) [reply files/2da/2da_bgee/statdesc.htm]: @lynxlynxlynx 

So, I ended up making an invisible `<table>`, i.e.: `<td style="border: 0; font-family: monospace;">`.

**Comment by 4Luke4** [comment]
@lynxlynxlynx 

Feel free to start reviewing.

**Comment by lynxlynxlynx** [comment]
I do it as the commits come, so it's more efficient. Thanks!

**Comment by 4Luke4** [comment]
> I do it as the commits come, so it's more efficient. Thanks!

I'll avoid flagging pull requests as draft then 😄 

Anyway, I'm very satisfied with the formatting now 👍🏻

**Pull request merged**